File: Programming/Practice/07/C++/07.cpp
#include <iostream>
#include <cmath>
#include <conio.h>
using namespace std;

int main(){
	setlocale(LC_ALL, "RUSSIAN");
	int opt;
	double a, b, c, S, p, AB, AB1, AB2, A1, A2, BC, BC1, BC2, B1, B2, CA, CA1, CA2, C1, C2;
	cout << "Выберите способ ввода параметров треугольника:" << endl;
	cin >> opt;
	if (int(opt) == 1){
		cout << "Введите длину стороны a:" << endl;
		cin >> a;
		cout << "Введите длину стороны b:" << endl;
		cin >> b;
		cout << "Введите длину стороны c:" << endl;
		cin >> c;
		// Вычисление площади треугольника по формуле Герона
		p = ((double(a) + double(b) + double(c)) / 2);
		S = (sqrt(double(p) * (double(p) - double(a)) * (double(p) - double(b)) * (double(p) - double(c))));
		cout << "S = ";
		cout << S << endl;
	}
	else if (int(opt) != 1 && int(opt) <= 1000 && int(opt) >= -1000){
		cout << "Введите координаты вершины A(x;y):" << endl;
		cin >> A1 >> A2;
		cout << "Введите координаты вершины B(x;y):" << endl;
		cin >> B1 >> B2;
		cout << "Введите координаты вершины C(x;y):" << endl;
		cin >> C1 >> C2;
		// Вычисление координат векторов сторон треугольника
		AB1 = double(B1) - double(A1);
		AB2 = double(B2) - double(A2);
		BC1 = double(C1) - double(B1);
		BC2 = double(C2) - double(B2);
		CA1 = double(A1) - double(C1);
		CA2 = double(A2) - double(C2);
		// Вычисление длин сторон треугольника
		AB = sqrt(AB1 * AB1 + AB2 * AB2);
		BC = sqrt(BC1 * BC1 + BC2 * BC2);
		CA = sqrt(CA1 * CA1 + CA2 * CA2);
		// Вычисление площади треугольника по формуле Герона
		p = ((AB + BC + CA) / 2);
		S = sqrt(p * (p - AB) * (p - BC) * (p - CA));
		cout << "S = ";
		cout << S << endl;
	}
	else if (int(opt) > 1000 || int(opt) < -1000){
	    cout << "Ошибочный ввод" << endl;
	}
	}

File: Programming/Practice/07/Python/07.py
from math import sqrt
print("Выберите способ ввода параметров треугольника: ")
opt = input()
if int(opt) == 1:
	print("Введите длину стороны a: ")
	a = input()
	print("Введите длину стороны b: ")
	b = input()
	print("Введите длину стороны c: ")
	c = input()
	# Вычисление площади треугольника по формуле Герона
	p = ((float(a)+float(b)+float(c))/2)
	S = sqrt(p*(p-float(a))*(p-float(b))*(p-float(c)))
	print("S = ", S)
elif int(opt) != 1 and (int(opt) >= -1000 and int(opt) <= 1000):
	print("Введите координаты вершины A(x;y):")
	A1, A2 = input()
	print("Введите координаты вершины B(x;y):")
	B1, B2 = input()
	print("Введите координаты вершины C(x;y):")
	C1, C2 = input()
	# Вычисление координат векторов сторон треугольника
	AB1 = float(B1) - float(A1)
	AB2 = float(B2) - float(A2)
	BC1 = float(C1) - float(B1)
	BC2 = float(C2) - float(B2)
	CA1 = float(A1) - float(C1)
	CA2 = float(A2) - float(C2)
	# Вычисление длин сторон трегольника по координатам их векторов
	AB = sqrt(AB1**2 + AB2**2)
	BC = sqrt(BC1**2 + BC2**2)
	CA = sqrt(CA1**2 + CA2**2)
	# Вычисление площади треугольника по формуле Герона
	p = (AB + BC + CA)/2
	S = sqrt(p*(p - AB)*(p - BC)*(p - CA))
	print("S = ", S)
elif int(opt) > 1000 or int(opt) < -1000:
	print("Ошибочный ввод")